pkg/asset/agent/mirror/mirror.go

```diff
@@ -55,3 +58,31 @@ func (m *AgentMirror) Load(f asset.FileFetcher) (bool, error) {
 
 	return false, nil
 }
+
+// GetReleaseImageMirror returns the mirror corresponding to the release image
+func (m *AgentMirror) GetReleaseImageMirror(releaseImage string) string {
+	// If registries.conf is defined, find mirror stripping the version out, e.g. for
+	// registry.ci.openshift.org/ocp/release:4.11.0-0.nightly-2022-06-06-201913, use
+	// registry.ci.openshift.org/ocp/release
+	source := strings.Split(":", releaseImage)
+
+	for _, file := range m.FileList {
+		if file.Filename == RegistriesConfFilename {
+			imageIndex := bytes.Index([]byte(source[0]), file.Data)
+			if imageIndex == -1 {
+				return ""
+			}
+			mirrorIndex := bytes.Index([]byte("[[registry.mirror]]"), file.Data[imageIndex+1:])
+			if mirrorIndex == -1 {
+				return ""
+			}
+			// get mirror from "location = xxx"
+			fields := bytes.Fields(file.Data[mirrorIndex+1:])
+			if len(fields) > 2 && string(fields[0]) == "location" {
+				return string(fields[2])
+			}
```

Comment:
As a personal preference, I'd find easier maybe to develop/use a standalone parser specific for such file

Reply:
Yes the function will now parse the entire file and store it in an array of structs with source and mirror.